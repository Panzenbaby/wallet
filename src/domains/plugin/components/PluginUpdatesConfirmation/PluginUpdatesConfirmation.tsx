import { Button } from "app/components/Button";
import { Image } from "app/components/Image";
import { Modal } from "app/components/Modal";
import { Table, TableCell, TableRow } from "app/components/Table";
import { usePluginIcon } from "domains/plugin/hooks/use-plugin-icon";
import React from "react";
import { useTranslation } from "react-i18next";

import { PluginImage } from "../PluginImage";

interface Properties {
	isOpen: boolean;
	plugins: any[];
	onClose?: () => void;
	onContinue?: () => void;
}

export const PluginUpdatesConfirmation = ({ isOpen, plugins, onClose, onContinue }: Properties) => {
	const { t } = useTranslation();

	const { renderPluginIcon } = usePluginIcon();

	const columns = [
		{
			Header: t("COMMON.NAME"),
			accessor: "name",
		},
		{
			Header: t("PLUGINS.REQUIRED_VERSION"),
			accessor: "minimumVersion",
			className: "justify-end",
		},
	];

	return (
		<Modal
			title={t("PLUGINS.MODAL_UPDATES_CONFIRMATION.TITLE")}
			image={<Image name="GenericWarning" className="m-auto my-8 w-3/5" />}
			description={t("PLUGINS.MODAL_UPDATES_CONFIRMATION.DESCRIPTION")}
			size="xl"
			isOpen={isOpen}
			onClose={onClose}
		>
			<div data-testid="PluginUpdatesConfirmation" className="mt-5">
				<Table data={plugins} columns={columns}>
					{(pluginData: any) => (
						<TableRow>
							<TableCell variant="start" innerClassName="space-x-4" style={{ width: "70%" }}>
								<PluginImage
									logoURL={pluginData.logo}
									isExchange={pluginData.category === "exchange"}
									size="sm"
								/>

								<div className="flex items-center space-x-2">
									<span
										data-testid="PluginUpdates__title"
										className="flex items-center font-semibold link important:px-0"
									>
										{pluginData.title}
									</span>

									{renderPluginIcon({
										isGrant: pluginData.isGrant,
										isOfficial: pluginData.isOfficial,
									})}
								</div>
							</TableCell>

							<TableCell variant="end" innerClassName="justify-end">
								<span data-testid="PluginUpdates__minimum-version">{pluginData.minimumVersion}</span>
							</TableCell>
						</TableRow>
					)}
				</Table>

				<div className="flex justify-end mt-8 space-x-3">
					<Button variant="secondary" onClick={onClose} data-testid="PluginUpdates__cancel-button">
						{t("COMMON.CANCEL")}
					</Button>

					<Button onClick={onContinue} data-testid="PluginUpdates__continue-button">
						<span>{t("COMMON.CONTINUE")}</span>
					</Button>
				</div>
			</div>
		</Modal>
	);
};

PluginUpdatesConfirmation.defaultProps = {
	isOpen: false,
};
