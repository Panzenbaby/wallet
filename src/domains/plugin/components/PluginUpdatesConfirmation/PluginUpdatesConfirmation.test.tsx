import React from "react";
import { render, screen } from "utils/testing-library";

import { PluginUpdatesConfirmation } from "./PluginUpdatesConfirmation";

describe("Plugin Updates Confirmation", () => {
	const plugins = [
		{
			isOfficial: true,
			logo: "https://payvo.com/logo.png",
			title: "ARK Explorer",
			updateStatus: {
				isAvailable: true,
				isCompatible: true,
				minimumVersion: "3.0.5",
			},
		},
		{
			title: "Animal Avatars",
			updateStatus: {
				isAvailable: true,
				isCompatible: false,
				minimumVersion: "3.1.0",
			},
		},
	];

	it("should render", () => {
		const { container } = render(<PluginUpdatesConfirmation isOpen plugins={plugins} />);

		expect(screen.getAllByRole("row")).toHaveLength(3);
		expect(screen.getByText("ARK Explorer")).toBeInTheDocument();
		expect(screen.getByText("3.1.0")).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
