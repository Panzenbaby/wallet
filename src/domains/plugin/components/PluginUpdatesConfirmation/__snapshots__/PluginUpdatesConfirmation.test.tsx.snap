// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin Updates Confirmation should render 1`] = `
<div>
  <div
    class="flex fixed inset-0 z-50 py-20 w-full h-full bg-opacity-60 dark:bg-opacity-80 bg-theme-secondary-900-rgba overflow-overlay dark:bg-black-rgba"
    data-testid="modal__overlay"
  >
    <div
      class="Modal__ModalContainer-sc-5lu2mc-0 bqQQPN"
      tabindex="-1"
    >
      <div
        class="flex overflow-hidden relative flex-col p-10 shadow-2xl rounded-2.5xl bg-theme-background"
        data-testid="modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded transition-all duration-100 ease-linear bg-theme-primary-100 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400 hover:bg-theme-primary-300"
        >
          <button
            class="OriginalButton__StyledButton-sc-o0fs2j-0 jWtMkt w-11 h-11"
            data-testid="modal__close-btn"
            type="button"
          >
            <div
              class="flex relative items-center space-x-2"
            >
              <div
                class="sc-bdnxRM AFVmm"
                height="16"
                width="16"
              >
                <svg>
                  cross-slim.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative space-y-4"
        >
          <h2
            class="mb-0 text-3xl font-bold"
          >
            Plugin Updates
          </h2>
          <div
            class="flex-1"
          >
            <svg
              class="m-auto my-8 w-3/5"
            >
              generic-warning-dark-green.svg
            </svg>
            <div
              class="whitespace-pre-line text-theme-secondary-text"
            >
              Some of the plugins cannot be updated as they require a newer version of the Payvo Wallet. Pressing Continue will update plugins that can be updated.
            </div>
            <div
              class="mt-5"
              data-testid="PluginUpdatesConfirmation"
            >
              <div
                class="Table__TableWrapper-sc-1gcp0fb-0 fDAVsO -mt-3"
                role="table"
              >
                <table
                  cellpadding="0"
                  class="table-auto"
                >
                  <thead>
                    <tr
                      class="border-b border-theme-secondary-300 dark:border-theme-secondary-800"
                      role="row"
                    >
                      <th
                        class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold hasBorder"
                        colspan="1"
                        data-testid="table__th--0"
                        role="columnheader"
                        style="cursor: pointer;"
                        title="Toggle SortBy"
                      >
                        <div
                          class="flex flex-inline align-top"
                        >
                          <div>
                            Name
                          </div>
                          <div
                            class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
                          >
                            <div
                              class="sc-bdnxRM AkeAK transition-transform transform rotate-180"
                              height="10"
                              role="img"
                              width="10"
                            >
                              <svg>
                                chevron-down.svg
                              </svg>
                            </div>
                          </div>
                        </div>
                      </th>
                      <th
                        class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold hasBorder"
                        colspan="1"
                        data-testid="table__th--1"
                        role="columnheader"
                        style="cursor: pointer;"
                        title="Toggle SortBy"
                      >
                        <div
                          class="flex flex-inline align-top justify-end flex-row-reverse"
                        >
                          <div>
                            Required Version
                          </div>
                          <div
                            class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
                          >
                            <div
                              class="sc-bdnxRM AkeAK transition-transform transform rotate-180"
                              height="10"
                              role="img"
                              width="10"
                            >
                              <svg>
                                chevron-down.svg
                              </svg>
                            </div>
                          </div>
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    role="rowgroup"
                  >
                    <tr
                      class="TableRow__TableRowStyled-sc-y9j4av-0 cPmOfx"
                      data-testid="TableRow"
                    >
                      <td
                        style="width: 70%;"
                      >
                        <div
                          class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jGsTbn space-x-4"
                        >
                          <div
                            class="PluginImage__PluginImageWrapper-sc-10kztrf-0 bdbYlY bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700"
                            data-testid="PluginImage"
                          >
                            <img
                              alt="Logo"
                              class="object-cover w-full h-full"
                              src="https://payvo.com/logo.png"
                            />
                          </div>
                          <div
                            class="flex items-center space-x-2"
                          >
                            <span
                              class="flex items-center font-semibold link important:px-0"
                              data-testid="PluginUpdates__title"
                            >
                              ARK Explorer
                            </span>
                            <div
                              class="sc-bdnxRM AFVmm text-theme-warning-600"
                              height="16"
                              width="16"
                            >
                              <svg>
                                official-plugin.svg
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td>
                        <div
                          class="TableCell__TableCellInnerWrapper-sc-v39gua-0 hNhRDD justify-end"
                        >
                          <span
                            data-testid="PluginUpdates__minimum-version"
                          >
                            3.0.5
                          </span>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="TableRow__TableRowStyled-sc-y9j4av-0 cPmOfx"
                      data-testid="TableRow"
                    >
                      <td
                        style="width: 70%;"
                      >
                        <div
                          class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jGsTbn space-x-4"
                        >
                          <div
                            class="PluginImage__PluginImageWrapper-sc-10kztrf-0 bdbYlY bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700"
                            data-testid="PluginImage"
                          >
                            <svg>
                              plugin-logo-placeholder.svg
                            </svg>
                          </div>
                          <div
                            class="flex items-center space-x-2"
                          >
                            <span
                              class="flex items-center font-semibold link important:px-0"
                              data-testid="PluginUpdates__title"
                            >
                              Animal Avatars
                            </span>
                          </div>
                        </div>
                      </td>
                      <td>
                        <div
                          class="TableCell__TableCellInnerWrapper-sc-v39gua-0 hNhRDD justify-end"
                        >
                          <span
                            data-testid="PluginUpdates__minimum-version"
                          >
                            3.1.0
                          </span>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="flex justify-end mt-8 space-x-3"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 jfKZOY"
                  data-testid="PluginUpdates__cancel-button"
                  type="button"
                >
                  <div
                    class="flex relative items-center space-x-2"
                  >
                    Cancel
                  </div>
                </button>
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 jVYfXJ"
                  data-testid="PluginUpdates__continue-button"
                  type="button"
                >
                  <div
                    class="flex relative items-center space-x-2"
                  >
                    <span>
                      Continue
                    </span>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
